File: ce/snippets/csharp/CRMV8/reporting/cs/downloadreportdefinition1.cs
// <snippetdownloadreportdefinition1>



                 // Query for an existing report: Account Overview. This is a default report in Microsoft Dynamics CRM.				    
                 QueryByAttribute reportQuery = new QueryByAttribute(Report.EntityLogicalName);
                 reportQuery.AddAttributeValue("name", "Account Overview");
                 reportQuery.ColumnSet = new ColumnSet("reportid");
                 
	    // Get the report.
	    EntityCollection retrieveReports = _serviceProxy.RetrieveMultiple(reportQuery);
	
	    // Convert retrieved Entity to a report
	    Report retrievedReport = (Report)retrieveReports.Entities[0];
                 Console.WriteLine("Retrieved the 'Account Overview' report.");
	
	    // Use the Download Report Definition message.
	    DownloadReportDefinitionRequest rdlRequest = new DownloadReportDefinitionRequest
	    {
                     ReportId = retrievedReport.ReportId.Value
                 };

	    DownloadReportDefinitionResponse rdlResponse = (DownloadReportDefinitionResponse)_serviceProxy.Execute(rdlRequest);

                 // Get the current directory path.
                 _currentDirectoryPath = Directory.GetCurrentDirectory();

	    // Access the xml data and save to disk
	    XmlTextWriter reportDefinitionFile = new XmlTextWriter( _currentDirectoryPath + "\\NewReport.rdl", System.Text.Encoding.UTF8);
	    reportDefinitionFile.WriteRaw(rdlResponse.BodyText);
	    reportDefinitionFile.Close();

                 if (File.Exists(_currentDirectoryPath + "\\NewReport.rdl"))
	    {
                     Console.WriteLine("Downloaded the report definition (NewReport.rdl) to '{0}'.", _currentDirectoryPath.ToString());
	    }

// </snippetdownloadreportdefinition1>

File: ce/snippets/visualbasic/CRMV8/reportingvb/vb/downloadreportdefinition1.vb
' <snippetdownloadreportdefinition1>



 ' Query for an existing report: Account Overview. This is a default report in Microsoft Dynamics CRM.                    
 Dim reportQuery As New QueryByAttribute(Report.EntityLogicalName)
 reportQuery.AddAttributeValue("name", "Account Overview")
 reportQuery.ColumnSet = New ColumnSet("reportid")

 ' Get the report.
 Dim retrieveReports As EntityCollection =
  _serviceProxy.RetrieveMultiple(reportQuery)

 ' Convert retrieved Entity to a report
 Dim retrievedReport As Report = CType(retrieveReports.Entities(0), Report)
 Console.WriteLine("Retrieved the 'Account Overview' report.")

 ' Use the Download Report Definition message.
 Dim rdlRequest As DownloadReportDefinitionRequest =
  New DownloadReportDefinitionRequest With {
   .ReportId = retrievedReport.ReportId.Value
  }

 Dim rdlResponse As DownloadReportDefinitionResponse =
  CType(_serviceProxy.Execute(rdlRequest), DownloadReportDefinitionResponse)

 ' Get the current directory path.
 _currentDirectoryPath = Directory.GetCurrentDirectory()

 ' Access the xml data and save to disk
 Dim reportDefinitionFile As New XmlTextWriter(_currentDirectoryPath &amp; "\NewReport.rdl", System.Text.Encoding.UTF8)
 reportDefinitionFile.WriteRaw(rdlResponse.BodyText)
 reportDefinitionFile.Close()

 If File.Exists(_currentDirectoryPath &amp; "\NewReport.rdl") Then
  Console.WriteLine("Downloaded the report definition (NewReport.rdl) to '{0}'.", _currentDirectoryPath.ToString())
 End If

' </snippetdownloadreportdefinition1>